test/local/account_routes.js

```diff
@@ -161,6 +165,104 @@ test(
 )
 
 test(
+  '/recovery_email/status with sign-in confirmation enabled',
+  function (t) {
+
+    var configOptions = {
+      signinConfirmation: {
+        enabled: true,
+        sample_rate: 1.0
+      }
+    }
+
+    var mockDB = {
+      deleteAccount: sinon.spy()
+    }
+
+    var mockRequest = {
+      auth: {
+        credentials: {
+          uid: uuid.v4('binary').toString('hex'),
+          email: TEST_EMAIL,
+          emailVerified: true,
+          tokenVerified: true
+        }
+      }
+    }
+
+    var accountRoutes = makeRoutes({
+      config: configOptions,
+      db: mockDB
+    })
+
+    var route = getRoute(accountRoutes, '/recovery_email/status')
+    return new P(function(resolve) {
+      route.handler(mockRequest, function(response) {
+        resolve(response)
+      })
+    })
+    .then(function(response) {
+      t.equal(mockDB.deleteAccount.callCount, 0)
```

Comment:
Having a spy to test that `deleteAccount` was never called seems a little odd, do we have reason to worry that it might be called?  If so, adding a comment here somewhere would be helpful.